homeassistant/components/lifx/manager.py

```diff
@@ -179,15 +211,33 @@ async def start_effect(
         self, entity_ids: set[str], service: str, **kwargs: Any
     ) -> None:
         """Start a light effect on entities."""
-        bulbs = [
-            coordinator.device
+
+        coordinators: list[LIFXUpdateCoordinator] = [
+            coordinator
             for entry_id, coordinator in self.hass.data[DOMAIN].items()
             if entry_id != DATA_LIFX_MANAGER
             and self.entry_id_to_entity_id[entry_id] in entity_ids
         ]
-        _LOGGER.debug("Starting effect %s on %s", service, bulbs)
 
-        if service == SERVICE_EFFECT_PULSE:
+        bulbs: list[Light] = [coordinator.device for coordinator in coordinators]
```

Comment:
It would be slightly more efficient to only enumerate `self.hass.data[DOMAIN].items()` once and build your coordinators and bulbs list in a loop instead of second list comprehension built from the first one.

It probably doesn't matter unless you have 100+ bulbs though